packages/block-editor/src/components/inserter/hooks/use-block-types-state.js

```diff
@@ -23,6 +23,26 @@ import { store as blockEditorStore } from '../../../store';
  * @return {Array} Returns the block types state. (block types, categories, collections, onSelect handler)
  */
 const useBlockTypesState = ( rootClientId, onInsert ) => {
+	const [ allItems ] = useSelect( ( select ) => {
+		const { getBlocksByName, getInserterItems } =
+			select( blockEditorStore );
+
+		// Try the empty root first.
+		const rootInserterItems = getInserterItems( '' );
+		if ( rootInserterItems.length ) {
+			return [ rootInserterItems ];
+		}
+
+		const postContentBlock = getBlocksByName( 'core/post-content' );
```

Comment:
Yea this adds a soft dependency to the WordPress block library in the agnostic block editor package. That should be another better way, ideally "if we can't find insertable items up let's look down until we can".

Reply:
Agreed. It'll probably be useful to move something like this into a selector in the block editor store like "getClosestInsertionPoint"

Reply:
for now I think something like:
```
const [ allItems ] = useSelect( ( select ) => {
               let availableItems = select( blockEditorStore ).getInserterItems( '' );
               // use current selection as root for situations like
               // template locked mode
               if ( availableItems ) {
                       const selectedBlock =
                               select( blockEditorStore ).getSelectedBlockClientId();
                       availableItems = select( blockEditorStore ).getInserterItems( selectedBlock );
               }
               return [ availableItems ];
       } );
```

works good enough.